raven-log4j/src/main/java/com/getsentry/raven/log4j/SentryAppender.java

```diff
@@ -110,6 +107,54 @@ public SentryAppender(Raven raven) {
     }
 
     /**
+     * Do some appender initialization *after* instance construction, so that we don't
+     * log in the constructor (which can cause annoying messages) and so that system
+     * properties and environment variables override hardcoded appender configuration.
+     */
+    @SuppressWarnings("checkstyle:hiddenfield")
+    private void lazyInit() {
```

Comment:
OK, I actually read the code this time. 😬 

I think what I was noting was mostly that this pattern seems to exist side-by-side in each usage:

```java
lazyInit();
if (raven == null) {
    initRaven();
}
```

It makes sense that is initialization check can't exist within the `if` block, but can't we combine these two methods somehow into a standard pattern for initializing settings if necessary, then instantiating the `Raven` instance if necessary, rather than having them independent like this? Or was there a reason the `initRaven` method had to be wrapped in the conditional (rather than containing the conditional) that I'm forgetting?